model/task_lifecycle.go

```diff
@@ -220,10 +225,21 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 			return errors.Wrap(err, "Error marking task as ended")
 		}
 	}
+	if t.IsPartOfSingleHostTaskGroup() {
+		if err = t.SetResetWhenFinished(); err != nil {
+			return errors.Wrapf(err, "error setting resetWhenFinished for task")
 
-	if err = resetTask(t.Id, user); err != nil {
-		return err
+		}
+		// want to restart all tasks in the task group, if it's finished
+		if err = tryResetTaskGroup(t); err != nil {
```

Comment:
Hmm I see what you mean. I meant to return the error here because an error needs to be returned to the UI ("task group is not finished" etc), but I can case on origin?